cob_twist_controller/src/cob_twist_controller.cpp

```diff
@@ -505,6 +520,12 @@ void CobTwistController::jointstateCallback(const sensor_msgs::JointState::Const
         last_q_ = q_temp;
         last_q_dot_ = q_dot_temp;
     }
+    initial_pos_.clear();
+
+    for(int i = 0; i< msg->position.size();i++)
+    {
+        initial_pos_.push_back(msg->position[i]);
```

Comment:
I think you should use `q_temp` or `last_q` here...as it is guaranteed to only contain the required joint values...